### PR TITLE
minhook: add minhook/cci.20240114 recipe

### PR DESCRIPTION
**Pull request by wu-vincent**
Specify library name and version:  **minhook/cci.20240114**

[**minhook**](https://github.com/TsudaKageyu/minhook) is a x86/x64 API Hooking Library for Windows. It is a lightweight alternative to the Detours library. Although I am not the original author, I believe creating a Conan recipe for this useful library will benefit the community with easier integration and broader usage.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Maybe we could ask the maintainers to also create a new release? The last one is from 2017!

**Comment by wu-vincent** [comment]
Hi, thank you for your suggestion. I completely agree that having the latest releases in Conan Center is ideal for ensuring the most up-to-date and stable versions are available. Someone raised an [issue](https://github.com/TsudaKageyu/minhook/issues/113) back in April to request a new release. Unfortunately, there hasn't been a response from the maintainer yet.

In light of this, I propose creating a cci version of the package (e.g., minhook/cci.20240108). This would enable users to benefit from the recent improvements while we await an official release. This approach has been adopted for other libraries in similar situations, such as [detours/cci.20220630](https://conan.io/center/recipes/detours?version=cci.20220630).

I'd appreciate your thoughts on this approach or any alternative suggestions you might have.

**Comment by RubenRBS** [comment]
@wu-vincent let's do that!

File: recipes/minhook/all/patches/cci.20240114-0001-increase-cmake-min.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index d45414e..533dac6 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -24,7 +24,7 @@
+ #  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ #  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+-cmake_minimum_required(VERSION 3.0)
++cmake_minimum_required(VERSION 3.5)
```
Comment (valgur) [inline recipes/minhook/all/patches/cci.20240114-0001-increase-cmake-min.patch:10]:
```suggestion
+cmake_minimum_required(VERSION 3.15)
```
Might as well make it 3.15, which avoids `shared` and `fPIC` issues and is required by Conan anyway.

**Review by valgur — CHANGES_REQUESTED** [review]
LGTM. Just one minor change required.

**Comment by wu-vincent** [comment]
@valgur Thank you for the suggestion! I've made the change accordingly.

**Review by valgur — APPROVED** [review]

File: recipes/minhook/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+sources:
+  "cci.20240114":
```
Comment (memsharded) [inline recipes/minhook/all/conandata.yml:2]: We will be updating this versioning scheme, to be more future-proof. I'll do it myself and contribute.

**Review by memsharded — COMMENTED** [review]
This is looking good, I'll try to give it a final push to get it merged.
Thanks for your contribution, and I apologize for the delays in reviewing, the load is simply overwhelming and prioritizing is very challenging.

**Review by AbrilRBS — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Pull request merged**